src/client/helpers/helpers.js

```diff
@@ -220,6 +220,59 @@ function constructTagList(data) {
   </li>`);
 }
 
+function saveMemoryIdToStorage(id) {
+  if (localStorage.getItem('toDelete') !== null) {
+    const memoriesWaitingToBeRemoved = JSON.parse(localStorage.getItem("toDelete"));
+    memoriesWaitingToBeRemoved.memories.push(id);
+    const saveMemoriesToDelete = JSON.stringify(memoriesWaitingToBeRemoved);
+    localStorage.toDelete = saveMemoriesToDelete;
+  }
+  else {
+    const saveDeletedMemory = JSON.stringify({memories: [id]});
+    localStorage.toDelete = saveDeletedMemory;
+  }
+}
+
+function removeMemoryFromStoredData(id) {
+  const offlineData = JSON.parse(localStorage.getItem('data'));
+  offlineData.forEach((memory, index) => {
+    if (memory.id == id) {
+      offlineData.splice(index, 1);
+    }
+  });
+  const offlineDataAfterRemoving = JSON.stringify(offlineData);
+  localStorage.data = offlineDataAfterRemoving;
+return offlineData;
+}
+
+function removeMemoryFromLocalStorage(index) {
+  const memoriesWaitingToBeRemoved = JSON.parse(localStorage.getItem("toDelete"));
+  if (memoriesWaitingToBeRemoved.memories.length === 1) {
+    localStorage.removeItem('toDelete');
+  }
+  else {
+  memoriesWaitingToBeRemoved.memories.splice(index, 1);
+  const memoriesStillToDelete = JSON.stringify(memoriesWaitingToBeRemoved);
+  localStorage.toDelete = memoriesStillToDelete;
+}
+}
+
+function removeMemoriesDeletedOffline(cb) {
+  if(localStorage.getItem('toDelete') !== null) {
+  const deletedMemories = JSON.parse(localStorage.getItem('toDelete'));
+  deletedMemories.memories.forEach((memory, index) => {
```

Comment:
This is bad..

You should use `async.parrallel`
https://caolan.github.io/async/docs.html#parallel

If you do not feel confident doing this dw